### PR TITLE
perf: Added individual tests

### PR DESCRIPTION
**Pull request by nasastry**
Instead of running group of tests and deciding whether they failed or not
at the end. With this patch tests run individually.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
Results on Power9:

JOB ID     : cbe42e389a52b4bd40715a4d47d898a082342f6b
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-22T02.47-cbe42e3/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (0.29 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (0.30 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: PASS (0.35 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: PASS (0.32 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: PASS (0.35 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: PASS (0.73 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (0.29 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (0.37 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (0.29 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (0.26 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 4.57 s

Results on Power8

JOB ID     : 04ff58d3fbe9d26557bdb75c9e9a233ceffcbf3b
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-22T02.47-04ff58d/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (1.55 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (1.49 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: PASS (1.44 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: PASS (2.44 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: PASS (2.43 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: PASS (6.70 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (1.47 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (1.49 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (1.38 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (1.36 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 23.59 s

**Comment by nasastry** [comment]
Figured other errors with the existing scripts. Will send a cleaned code. Please don't merge this yet.

**Comment by nasastry** [comment]
Updated the code, please review.

Here are the results on Power8 and Power9 machine.

Power9:
```
JOB ID     : 984dbf8994e10476efa73dac5642e9fe738a7fa4
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-22T06.18-984dbf8/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (0.25 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (0.33 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: CANCEL: Not supported on Power9 (0.24 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: CANCEL: Not supported on Power9 (0.26 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: CANCEL: Not supported on Power9 (0.24 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: CANCEL: Not supported on Power9 (0.27 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (0.25 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (0.41 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (0.27 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (0.30 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB TIME   : 3.92 s
```

Power8:
```
JOB ID     : f7479faece77e8944134bd9deb0f25409d2ad6c6
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-22T06.17-f7479fa/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (1.36 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (1.45 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: PASS (1.42 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: PASS (2.47 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: PASS (2.41 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: PASS (6.73 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (1.35 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (1.52 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (1.37 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (1.37 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 22.59 s
```

**Comment by PraveenPenguin** [comment]
@nasastry can you please split this patch in two commits as 
1- tests run individually.
2- skip test which is not applicable to p9 as it give false result

**Comment by nasastry** [comment]
@PraveenPenguin Split into two commits, Please review.

On Power8 machine

```
JOB ID     : a3b1d6bf30f5e3ee2420ff2bbd77cb06e573d688
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-25T10.24-a3b1d6b/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (2.57 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (2.98 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: PASS (2.99 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: PASS (3.94 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: PASS (3.68 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: PASS (9.11 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (2.35 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (3.06 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (2.33 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (2.33 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 36.83 s
```

On Power9 machine with first patch:

```
JOB ID     : 021ef5fdedd086b0979fc24163772a447cc5e07e
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-25T10.18-021ef5f/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (1.17 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (1.47 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: FAIL: Domain is not specified, perf unableto recognize it has invalid event (1.51 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: FAIL: perf unable to recognize domain namein param=value format (1.53 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: FAIL: domain does not exist but perf listedhas supported (1.49 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: FAIL: perf unable to recognize domain name inparam=value format for all domains (1.50 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (1.19 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (1.77 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (1.18 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (1.14 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 4 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 15.29 s
```

On Power9 machine with second patch:

```
JOB ID     : 29484e19e1e8287c504739f2692fcae22220e703
JOB LOG    : /home/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2019-03-25T10.22-29484e1/job.log
 (01/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_display_domain_indices_in_sysfs: PASS (1.17 s)
 (02/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_phys_core_param: PASS (1.63 s)
 (03/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_domain_param: CANCEL: Not supported on Power9 (1.13 s)
 (04/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_domain_param: CANCEL: Not supported on Power9 (1.12 s)
 (05/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_domain_not_existing: CANCEL: Not supported on Power9 (1.11 s)
 (06/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_check_all_domains: CANCEL: Not supported on Power9 (1.13 s)
 (07/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_w_chip_param: PASS (1.17 s)
 (08/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_wo_chip_param: PASS (1.82 s)
 (09/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_phys_core: PASS (1.16 s)
 (10/10) perf_24x7_hardware_counters.py:test_eliminate_domain_suffix.test_event_helper_vcpu: PASS (1.15 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB TIME   : 13.94 s
```

**Comment by PraveenPenguin** [comment]
@nasastry Travis complaint about pep8 for second commit , can you run auotpep -i <py file > and resolved pep8 issue and do force push , that make travis happy :)

**Comment by nasastry** [comment]
> @nasastry Travis complaint about pep8 for second commit , can you run auotpep -i and resolved pep8 issue and do force push , that make travis happy :)

@PraveenPenguin Done.

File: perf/perf_24x7_hardware_counters.py [hunk 1]
```diff
@@ -128,100 +96,103 @@ def display_domain_indices_in_sysfs(self):
...
         else:
             self.log.info('perf recognized unsupported event')
 
-    def event_w_domain_param(self):
+    def test_event_w_domain_param(self):
+        if self.cpu_arch == 'power9':
+            self.cancel("Not supported on Power9")
         result1 = self.event_stat(',domain=2,core=1/ sleep 1')
-        print result1.stderr
+        print(result1.stderr)
```
Comment (PraveenPenguin) [inline perf/perf_24x7_hardware_counters.py]: i think not needed here can you please remove this

**Comment by nasastry** [comment]
> @nasastry over all looks good , only one style issue as print statement can be removed , i will take care while merging is that fine ?

@PraveenPenguin I am fine with it.

**Pull request merged**